File: LeetCode/param.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function (s) {
    const hashBracket = { "(": ")", "{": "}", "[": "]" };
    const stack = [];
    for (let ele of s) {
      if (hashBracket[ele]) {
        stack.push(hashBracket[ele]);
      } else if (stack.length > 0 && stack[stack.length - 1] === ele) {
        stack.pop();
      } else {
        return false;
      }
    }
    return stack.length === 0;
  };

   console.log(isValid("[[[]"))

File: LeetCode/param.py
class Solution:
    def isValid(self, s: str) -> bool:
        hashBracket = { "(": ")", "{": "}", "[": "]" }
        stack = []
        for ele in s:
            if(ele in hashBracket):
                stack.append(hashBracket[ele])
            else:
                if(len(stack) > 0 and stack[len(stack) - 1] == ele):
                    stack.pop();  
                else:
                    return False
        return len(stack)==0
                      

 
p1 = Solution()
print(p1.isValid("[[[]"))
